### PR TITLE
bug: handle the logging of malformed URL exception

### PR DESCRIPTION
**Pull request by sarthak77**
As part of this PR,
- wrapped the malformed URL exception as a warning message
- refactored getHttpUrlFromRawAttributes for ease of readability

### CODE REVIEW

File: semantic-convention-utils/src/main/java/org/hypertrace/semantic/convention/utils/http/HttpSemanticConventionUtils.java [hunk 1]
```diff
@@ -431,13 +431,17 @@ public static URL getNormalizedUrl(String url) throws MalformedURLException {
       }
     }
 
+    if (httpUrlFromRawAttributes != null && isAbsoluteUrl(httpUrlFromRawAttributes)) {
```
Comment (kotharironak) [inline semantic-convention-utils/src/main/java/org/hypertrace/semantic/convention/utils/http/HttpSemanticConventionUtils.java:434]: It same logic, right? It is refactored?

Comment (sarthak77) [reply semantic-convention-utils/src/main/java/org/hypertrace/semantic/convention/utils/http/HttpSemanticConventionUtils.java:438]: I think we missed some conditions in previous way

File: semantic-convention-utils/src/main/java/org/hypertrace/semantic/convention/utils/http/HttpSemanticConventionUtils.java [hunk 2]
```diff
@@ -317,7 +321,7 @@ public static URL getNormalizedUrl(String url) throws MalformedURLException {
       try {
         return Optional.of(getNormalizedUrl(url.get()).getAuthority());
       } catch (MalformedURLException e) {
-        e.printStackTrace();
+        LOGGER.warn("Received invalid URL host : {}", e.getMessage());
```
Comment (kotharironak) [inline semantic-convention-utils/src/main/java/org/hypertrace/semantic/convention/utils/http/HttpSemanticConventionUtils.java:324]:
```suggestion
       LOGGER.warn("On extracting, httpHost received an invalid URL: {}, {}", url.get(), e.getMessage());
```

File: semantic-convention-utils/src/main/java/org/hypertrace/semantic/convention/utils/http/HttpSemanticConventionUtils.java [hunk 3]
```diff
@@ -335,7 +339,7 @@ public static URL getNormalizedUrl(String url) throws MalformedURLException {
         }
         return Optional.of(removeTrailingSlash(pathval));
       } catch (MalformedURLException e) {
-        e.printStackTrace();
+        LOGGER.warn("Received invalid URL path : {}", e.getMessage());
```
Comment (kotharironak) [inline semantic-convention-utils/src/main/java/org/hypertrace/semantic/convention/utils/http/HttpSemanticConventionUtils.java:342]:
```suggestion
               LOGGER.warn("On extracting, httpPath received an invalid URL: {}, {}", url.get(), e.getMessage());

```

File: semantic-convention-utils/src/main/java/org/hypertrace/semantic/convention/utils/http/HttpSemanticConventionUtils.java [hunk 4]
```diff
@@ -382,7 +386,7 @@ public static URL getNormalizedUrl(String url) throws MalformedURLException {
       try {
         return Optional.of(getNormalizedUrl(url.get()).getProtocol());
       } catch (MalformedURLException e) {
-        e.printStackTrace();
+        LOGGER.warn("Received invalid URL scheme : {}", e.getMessage());
```
Comment (kotharironak) [inline semantic-convention-utils/src/main/java/org/hypertrace/semantic/convention/utils/http/HttpSemanticConventionUtils.java:389]:
```suggestion
      LOGGER.warn("On extracting, httpSchem received an invalid URL: {}, {}", url.get(), e.getMessage());
```

File: semantic-convention-utils/src/main/java/org/hypertrace/semantic/convention/utils/http/HttpSemanticConventionUtils.java [hunk 5]
```diff
@@ -451,7 +459,7 @@ public static URL getNormalizedUrl(String url) throws MalformedURLException {
       try {
         return Optional.ofNullable(getNormalizedUrl(url.get()).getQuery());
       } catch (MalformedURLException e) {
-        e.printStackTrace();
+        LOGGER.warn("Received invalid URL querystring : {}", e.getMessage());
```
Comment (kotharironak) [inline semantic-convention-utils/src/main/java/org/hypertrace/semantic/convention/utils/http/HttpSemanticConventionUtils.java:462]:
```suggestion
        LOGGER.warn("On extracting, httpQueryString received an invalid URL: {}, {}", url.get(), e.getMessage());
```

File: semantic-convention-utils/src/main/java/org/hypertrace/semantic/convention/utils/http/HttpSemanticConventionUtils.java [hunk 6]
```diff
@@ -517,7 +525,7 @@ public static int getHttpResponseStatusCode(Event event) {
       URL url = getNormalizedUrl(urlPath);
       return Optional.of(url.getPath());
     } catch (MalformedURLException e) {
-      e.printStackTrace();
+      LOGGER.warn("Received invalid URL path : {}, {}", urlPath, e.getMessage());
```
Comment (kotharironak) [inline semantic-convention-utils/src/main/java/org/hypertrace/semantic/convention/utils/http/HttpSemanticConventionUtils.java:528]:
```suggestion
      LOGGER.warn("On extracting, httpResponseStatusCode received invalid URL path : {}, {}", urlPath, e.getMessage());
```

**Comment by kotharironak** [comment]
@sarthak77 pl. add desciption.

File: semantic-convention-utils/src/main/java/org/hypertrace/semantic/convention/utils/http/HttpSemanticConventionUtils.java (hunk 4 above)
Comment (rish691) [reply semantic-convention-utils/src/main/java/org/hypertrace/semantic/convention/utils/http/HttpSemanticConventionUtils.java:389]: nit: `scheme`

File: semantic-convention-utils/src/main/java/org/hypertrace/semantic/convention/utils/http/HttpSemanticConventionUtils.java (hunk 2 above)
Comment (rish691) [reply semantic-convention-utils/src/main/java/org/hypertrace/semantic/convention/utils/http/HttpSemanticConventionUtils.java:324]: nit: comma after httpHost

File: semantic-convention-utils/src/main/java/org/hypertrace/semantic/convention/utils/http/HttpSemanticConventionUtils.java (hunk 3 above)
Comment (rish691) [reply semantic-convention-utils/src/main/java/org/hypertrace/semantic/convention/utils/http/HttpSemanticConventionUtils.java:342]: nit: comma after httpPath

File: semantic-convention-utils/src/main/java/org/hypertrace/semantic/convention/utils/http/HttpSemanticConventionUtils.java (hunk 4 above)
Comment (rish691) [reply semantic-convention-utils/src/main/java/org/hypertrace/semantic/convention/utils/http/HttpSemanticConventionUtils.java:389]: nit: comma after httpScheme

File: semantic-convention-utils/src/main/java/org/hypertrace/semantic/convention/utils/http/HttpSemanticConventionUtils.java (hunk 5 above)
Comment (rish691) [reply semantic-convention-utils/src/main/java/org/hypertrace/semantic/convention/utils/http/HttpSemanticConventionUtils.java:462]: nit: comma after httpQueryString

File: semantic-convention-utils/src/main/java/org/hypertrace/semantic/convention/utils/http/HttpSemanticConventionUtils.java (hunk 6 above)
Comment (rish691) [reply semantic-convention-utils/src/main/java/org/hypertrace/semantic/convention/utils/http/HttpSemanticConventionUtils.java:528]: nit: comma after httpResponseStatusCode

File: semantic-convention-utils/src/main/java/org/hypertrace/semantic/convention/utils/http/HttpSemanticConventionUtils.java (hunk 1 above)
Comment (kotharironak) [reply semantic-convention-utils/src/main/java/org/hypertrace/semantic/convention/utils/http/HttpSemanticConventionUtils.java:438]: which condition?

Comment (sarthak77) [reply semantic-convention-utils/src/main/java/org/hypertrace/semantic/convention/utils/http/HttpSemanticConventionUtils.java:441]: yes got confused. Looks just refactoring

**Review by kotharironak — APPROVED** [review]

**Review by rish691 — APPROVED** [review]

**Pull request merged**